### ISSUE TITLE
not all attributes?

### ISSUE THREAD

**Issue by jgranduel** [jsumners/node-activedirectory#18]
Hi again, 

I've compared the results I get with findUser and PowerShell cmdlet Get-ADuser. Surprisingly I don't get all the properties with activedirectory (activedirectory2, but same issue with Gheeres').

I thought the attribute value might not be set, but that's not the case. Some are very useful like, say, 'officePhone'.
Why can't I query all attributes? is this a real issue or is there a way to get all values?

```
ad.findUser({attributes: ["cn", "dn", "officePhone"]}, "anAccoutName").then(console.log).catch(console.error)
> User {
  cn: 'First LastName',
  dn: 'CN=First LastName,OU=... ...DC=local' }
  
Get-ADUser anAccoutName -Properties * | select 'cn', "officePhone"

cn               officePhone
--               -----------
First LastName   0123452436...
```

Thanks

**Comment by jsumners** [comment]
Did you issue both queries with the same user to the same URL?

**Comment by jgranduel** [comment]
>Did you issue both queries with the same user to the same URL?

yes, same user, same URL.
PowerShell> Get-ADUser auser -Properties * -Credential $credentials -Server "the.server" | Get-Member -MemberType property | select name | measure
150

I don't know how to list all props with activedirectory2 (opts = {} should work, I thought, but it returns in my case only 13 attributes).

BTW, for the same user, 
```
PowerShell>Get-ADUser auser -Credential $creds -Server "the.server" | Get-Member -MemberType property | select -ExpandProperty name | sort
10 properties: DistinguishedName, Enabled, GivenName, Name, ObjectClass, ObjectGUID, SamAccountName, SID, Surname, UserPrincipalName
```
but 
```
activedirectory2 Node>new AD(config).findUser("auser").then(x=>console.log(Object.keys(x).sort().join(", "))).catch(console.error)
13 properties: cn, displayName, distinguishedName, dn, givenName, initials, mail, pwdLastSet, sAMAccountName, sn, userAccountControl, userPrincipalName, whenCreated
```
What's the reason of this difference? could it be changed if needed?

Thanks for any answer.

**Comment by jsumners** [comment]
If you pass an empty options object then the default configuration will be used. The default is to only pull a limited set of attributes; this saves query time and memory. As for why it's not pulling back your explicitly requested `officePhone` attribute, I'm not sure. It could be a bug or it could be something with the way Active Directory does things. I have seen this happen when the querying user doesn't have permissions to the attribute and when the attribute is located on a different host than the one queried.

You might also try using https://www.npmjs.com/package/adldap

**Comment by jgranduel** [comment]
Thanks, 
well, I thought about adldap --before realising you were the author.
When I read:
>This is a simple wrapper around ldapjs for basic operations against an Active Directory instance. If >you're looking for a robust interface, you may be interested in activedirectory2.
...I decided to go with activedirectory2.
No idea how to get or at least try something for getting missing property?

**Comment by jsumners** [comment]
But are you really using/planning to use things that this module does which adldap doesn't?

One thing you could do is supply a [Pino](http://getpino.io/) instance with the level set to 'trace' to see what sort of information gets logged. But you should also set a breakpoint at https://github.com/jsumners/node-activedirectory/blob/jsumners/lib/components/search.js#L195 and inspect the object returned by the underlying ldapjs library.

Sadly, I can't debug this on my own because I don't have a server that will exhibit the problem.

**Comment by ofaltins** [comment]
@jgranduel this should work: https://github.com/jsumners/node-activedirectory/blob/jsumners/test/attributes.js#L49

TLDR; Pass an asterisk as attributes in opts like so:
```
const opts = {
      attributes: {
        user: [ '*' ]
      }
}
```

**Comment by jgranduel** [comment]
Sorry for the very late response.
Indeed @ofaltins' answer worked out well (still not as many props as with PowerShell, but I haven't searched deeply).
Thanks

**Comment by ofaltins** [comment]
No worries!

**Comment by jsumners** [comment]
@jgranduel did you add a logger set to the `trace` level? That would be very helpful.
